pkg/scalers/ibmmq_scaler.go

```diff
@@ -0,0 +1,229 @@
+		meta.targetQueueDepth = defaultTargetQueueDepth
+	}
+
+	if val, ok := config.TriggerMetadata["tls"]; ok {
+		tlsDisabled, err := strconv.ParseBool(val)
+		if err != nil {
+			return nil, fmt.Errorf("invalid tls setting: %s", err)
+		}
+		meta.tlsDisabled = tlsDisabled
+	} else {
+		fmt.Println("No tls setting defined - setting default")
+		meta.tlsDisabled = defaultTLSDisabled
+	}
+
+	if val, ok := config.AuthParams["username"]; ok {
+		meta.username = val
+	} else {
+		return nil, fmt.Errorf("no username given")
+	}
+
+	if val, ok := config.AuthParams["password"]; ok {
+		meta.password = val
+	} else {
+		return nil, fmt.Errorf("no password given")
+	}
+
+	return &meta, nil
+}
+
+// IsActive returns true if there are messages to be processed/if we need to scale from zero
+func (s *IBMMQScaler) IsActive(ctx context.Context) (bool, error) {
+	queueDepth, err := s.getQueueDepthViaHTTP()
+	if err != nil {
+		return false, fmt.Errorf("error inspecting IBM MQ queue depth: %s", err)
+	}
+	return queueDepth > 0, nil
+}
+
+// getQueueDepthViaHTTP returns the depth of the MQ Queue from the Admin endpoint
+func (s *IBMMQScaler) getQueueDepthViaHTTP() (int, error) {
+	queue := s.metadata.queueName
+	url := s.metadata.host
+
+	var requestJSON = []byte(`{"type": "runCommandJSON", "command": "display", "qualifier": "qlocal", "name": "` + queue + `", "responseParameters" : ["CURDEPTH"]}`)
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer(requestJSON))
+	if err != nil {
+		return 0, fmt.Errorf("failed to request queue depth: %s", err)
+	}
+	req.Header.Set("ibm-mq-rest-csrf-token", "value")
+	req.Header.Set("Content-Type", "application/json")
+	req.SetBasicAuth(s.metadata.username, s.metadata.password)
+
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: s.metadata.tlsDisabled},
+	}
+	client := &http.Client{Transport: tr}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return 0, fmt.Errorf("failed to contact MQ via REST: %s", err)
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return 0, fmt.Errorf("failed to ready body of request: %s", err)
+	}
+
+	var response CommandResponse
+	err = json.Unmarshal(body, &response)
+	if err != nil {
+		return 0, fmt.Errorf("failed to parse JSON: %s", err)
+	}
+
+	if response.CommandResponse == nil || len(response.CommandResponse) == 0 {
+		return 0, fmt.Errorf("failed to parseresponse from REST call: %s", err)
+	}
+	return response.CommandResponse[0].Parameters.Curdepth, nil
+}
+
+// GetMetricSpecForScaling returns the MetricSpec for the Horizontal Pod Autoscaler
+func (s *IBMMQScaler) GetMetricSpecForScaling() []v2beta2.MetricSpec {
+	targetQueueLengthQty := resource.NewQuantity(int64(s.metadata.targetQueueDepth), resource.DecimalSI)
+	externalMetric := &v2beta2.ExternalMetricSource{
+		Metric: v2beta2.MetricIdentifier{
+			Name: kedautil.NormalizeString(fmt.Sprintf("%s-%s", "IBMMQ", s.metadata.queueName)),
```

Comment:
The metric name should be unique, that means that if you specify two or more IBM MQ scalers in one ScaledObject, each scaler should have a different metric name. I wonder if this is enough?
Is viable a usecase, where we have two MQ scalers with the same queueName but pointing to the different MQ (server)?

Reply:
This is a great point @zroubalik and would be a use case for sure, we are adding a solution now where we include the Queue Manager name in the metric name as well as the name of the Queue